django/contrib/postgres/search.py

```diff
@@ -72,11 +78,9 @@ class SearchVector(SearchVectorCombinable, Func):
     output_field = SearchVectorField()
     config = None
 
-    def __init__(self, *expressions, **extra):
-        super().__init__(*expressions, **extra)
-        config = self.extra.get('config', self.config)
-        self.config = SearchConfig(config) if config else None
-        weight = self.extra.get('weight')
+    def __init__(self, *expressions, config=None, weight=None):
+        super().__init__(*expressions)
+        self.config = SearchConfig.from_parameter(config)
```

Comment:
This ensures that `config` passed by `SearchVectorExact.process_rhs` ([and soon `SearchLookup.process_lhs`](https://code.djangoproject.com/ticket/31304#comment:4)) don't end up being nested and result in `'english'::regconfig::regconfig` in their final SQL.